Add QUIRK_MFM_CLOCK

TimothyPMann opened the pull request:
This quirk was needed to read some old UDOS 1526 V.3 or V.2 disks, which have clock bits set in certain places that violate the MFM clocking rules.

TimothyPMann committed:
Add QUIRK_MFM_CLOCK

This quirk was needed to read some old UDOS 1526 V.3 or V.2 disks,
which have clock bits set in certain places that violate the MFM
clocking rules.

qbarnes commented on ChangeLog:
```diff
@@ -1,3 +1,7 @@
+* Add QUIRK_MFM_CLOCK.  This quirk was needed to read some old UDOS
+  1526 V.3 or V.2 disks, which have clock bits set in certain places
```
Any reason why the "V"'s numbers are in reverse ascending order?  Both here in the commit log you have "V.3 or V.2" instead of "V.2 or V.3".  Unless there's a reason, it runs afoul of the element of least surprise.

Otherwise, all the changes look straightforward and good.  You even remembered to change the parselog which I would have forgotten about without your delta.  :)

TimothyPMann commented:
Rüdiger thought the disks were V.3 but wasn't sure. He thought possibly
they might be V.2. I guess I could explain that in more detail, but it is
all pretty vague.  I gather the disks aren't marked clearly.

On Fri, Nov 25, 2022, 10:52 AM Quentin Barnes ***@***.***>
wrote:

> ***@***.**** approved this pull request.
> ------------------------------
>
> In ChangeLog
> <https://github.com/qbarnes/cw2dmk/pull/27#discussion_r1032649314>:
>
> > @@ -1,3 +1,7 @@
> +* Add QUIRK_MFM_CLOCK.  This quirk was needed to read some old UDOS
> +  1526 V.3 or V.2 disks, which have clock bits set in certain places
>
> Any reason why the "V"'s numbers are in reverse ascending order? Both here
> in the commit log you have "V.3 or V.2" instead of "V.2 or V.3". Unless
> there's a reason, it runs afoul of the element of least surprise.
>
> Otherwise, all the changes look straightforward and good. You even
> remembered to change the parselog which I would have forgotten about
> without your delta. :)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/qbarnes/cw2dmk/pull/27#pullrequestreview-1194714917>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AILDEHIIE5AB2D7ZI5XJMN3WKEDH5ANCNFSM6AAAAAASLDSUOM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

qbarnes commented:
Would a better way to phrase it would be changing "V.3 or V.2 disks" to "(unsure if V.3 or possibly V.2 disks)"?

TimothyPMann commented:
Something like that. I will go rephrase it ...

On Sat, Nov 26, 2022, 11:11 AM Quentin Barnes ***@***.***>
wrote:

> Would a better way to phrase it would be changing "V.3 or V.2 disks" to
> "(unsure if V.3 or possibly V.2 disks)"?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/qbarnes/cw2dmk/pull/27#issuecomment-1328099308>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AILDEHMREJTVRU5Q4GRIWA3WKJOFBANCNFSM6AAAAAASLDSUOM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

TimothyPMann committed:
Improve ChangeLog comment about QUIRK_MFM_CLOCK

This quirk was needed to read some older UDOS disks with clock bits
set in certain places that violate the MFM clocking rules.  The most
frequent violation seen on a sample disk was that in a sequence of
bytes with data value ff ff ff..., the clocking pattern could be 00 88
88... instead of 00 00 00...  This results in flux transitions in
adjacent bit cells, which is normally illegal in MFM.  The disks were
probably from UDOS 1526 V.3, or possibly V.2.

TimothyPMann commented:
Should we do a release? I don't have any other changes that are anywhere near ready for prime time.

qbarnes commented:
Yes.  New release now available.

On Sat, Nov 26, 2022 at 12:21:03PM -0800, Timothy Mann wrote:
> Should we do a release? I don't have any other changes that are anywhere
> near ready for prime time.
>
> --
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because your review was requested.Message ID:
> ***@***.***>

TimothyPMann commented:
Thanks!

On Sun, Nov 27, 2022, 12:20 PM Quentin Barnes ***@***.***>
wrote:

> Yes. New release now available.
>
> On Sat, Nov 26, 2022 at 12:21:03PM -0800, Timothy Mann wrote:
> > Should we do a release? I don't have any other changes that are anywhere
> > near ready for prime time.
> >
> > --
> > Reply to this email directly, view it on GitHub, or unsubscribe.
> > You are receiving this because your review was requested.Message ID:
> > ***@***.***>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/qbarnes/cw2dmk/pull/27#issuecomment-1328331851>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AILDEHNXPHUUPYZBRASYDN3WKO7AHANCNFSM6AAAAAASLDSUOM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
